Sources/ArgumentParser/Usage/MessageInfo.swift

```diff
@@ -110,6 +121,8 @@ enum MessageInfo {
         self = .other(message: error.localizedDescription, exitCode: Int32(error.errorCode))
       case let error as LocalizedError where error.errorDescription != nil:
         self = .other(message: error.errorDescription!, exitCode: EXIT_FAILURE)
+      case let error as NSError:
+        self = .other(message: error.localizedDescription, exitCode: EXIT_FAILURE)
       default:
         self = .other(message: String(describing: error), exitCode: EXIT_FAILURE)
```

Comment:
Here we have a problem — on macOS, the `error as NSError` cast always succeeds, which means we lose the customization for non-`NSError` errors that conform to `CustomStringConvertible`. Since `NSError`, in turn, conforms to `CustomStringConvertible`, there's no way to distinguish them by casting alone.

This is an unfortunate but successful way of distinguishing actual `NSError` errors from plain Swift errors:
```suggestion
      default:
        if Swift.type(of: error) is NSError.Type, let error = error as? NSError {
          self = .other(message: error.localizedDescription, exitCode: EXIT_FAILURE)
        } else {
          self = .other(message: String(describing: error), exitCode: EXIT_FAILURE)
        }
```

Reply:
@natecook1000 interesting, and as always thanks for the amazing insight!  I'm wondering if the conditional cast to `NSError` needs to happen here for any reason then, or if a13105c would suffice?

Reply:
Does that compile? I don’t think `localizedDescription` will be available on `error` until the cast succeeds. Also note that this bridging behavior isn’t consistent between macOS and Linux!

Reply:
It does compile!  In fact Xcode gave me a warning about the cast always succeeding. I'm just making sure I didn't miss anything, but I'm happy to add it in if it's necessary.

Reply:
TIL! Let’s run with it :shipit: